vvm/utils/versioning.py

```diff
@@ -2,62 +2,62 @@
 import re
 from typing import Any, Optional
 
-from packaging.specifiers import Specifier
+from packaging.specifiers import SpecifierSet
 from packaging.version import Version
 
 from vvm.exceptions import UnexpectedVersionError
 from vvm.install import get_installable_vyper_versions, get_installed_vyper_versions
 
-_VERSION_RE = re.compile(r"\s*#\s*(?:pragma\s+|@)version\s+([=><^~]*)(\d+\.\d+\.\d+\S*)")
+_VERSION_RE = re.compile(r"^(?:#\s*(?:@version|pragma\s+version)\s+(.*))", re.MULTILINE)
```

Comment:
i think this misses pragmas which are indented for whatever reason. (actually pragmas can be anywhere in a file, but it's probably acceptable for vvm to assume they are the beginning)

Reply:
```suggestion
_VERSION_RE = re.compile(r"(?:#\s*(?:@version|pragma\s+version)\s+(.*))$", re.MULTILINE)
```

Reply:
Or what about this one to avoid cases like that?

```python
x:constant(String[30]) = "# @pragma version 0.3.1"
# pragma version 0.4.0"
@external
def foo():

    pass
```

```suggestion
_VERSION_RE = re.compile(r"^\s*(?:#\s*(?:@version|pragma\s+version)\s+(.*))", re.MULTILINE)
```

Reply:
the pragma could also be inside a triple-quoted string, e.g.
```
s: constant(String[256]) = """
# pragma version 0.3.10
"""
```
so long as we are not actually tokenizing the source, we run into these types of issues

Reply:
For future reference, this kind of contracts are annoying: 

```
x:constant(String[100]) = """
# pragma version 0.3.0
"""

# pragma version 0.4.0
```